vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/screenshot/ScreenCapturer.kt

```diff
@@ -31,9 +31,16 @@ class ScreenCapturer(
     suspend fun start() = coroutineScope {
         val outputStream = FileImageOutputStream(fileManager.createFile(FileType.SCREENSHOT, poolId, device.toDeviceInfo(), test))
         val writer = GifSequenceWriter(outputStream, TYPE_INT_ARGB, DELAY, true)
+        var targetOrientation = PORTRAIT
         while (isActive) {
             val capturingTimeMillis = measureTimeMillis {
-                getScreenshot()?.let { writer.writeToSequence(it) }
+                getScreenshot(targetOrientation)?.let {
+                    if (targetOrientation == UNDEFINED) {
```

Comment:
Look like that this condition will be always ```false```.
Maybe you wanted to define ```var targetOrientation = PORTRAIT``` to ```UNDEFINED``` ?

Reply:
Great catch. It should be actually UNDEFINED as long as it is unknown which orientation the images will have. based on the first screenshot we retrieve